src/blueprints/authentication.py

```diff
@@ -151,90 +151,91 @@ def isAdmin():
-#             return jsonify({
-#                 'success': False,
-#                 'message': 'expired'
-#             })
-
-#         if not pbkdf2_sha256.verify(one_time_password, passwordResetInfo.one_time_password):
-#             return jsonify({
-#                 'success': False,
-#                 'message': 'otp'
-#             })
-
-#         passwordResetInfo.has_reset = True
-#         user.set_password(new_password)
-
-#     except Exception as e:
-#         print(e)
-#         session.rollback()
-#         return jsonify({
-#             'success': False,
-#             'message': 'unknown'
-#         })
-
-#     else:
-#         session.commit()
-#         return jsonify({
-#             'success': True
-#         })
+@authentication.route('/reset-password', methods=["GET"])
+def request_reset():
+    try:
+        email = request.get_json()["email"]
+
+        user_query = session.query(User).filter(User.email == email)
+
+        if not user_query[0]:
+            return jsonify({
+                'success': False,
+                'error': 'email',
+                'message': 'User does not exist',
+            })
+
+        random_sequence = [str(math.floor(random.random() * 10)) for _ in range(6)]
+        otp = ''.join(random_sequence)
+
+        hashed_otp = pbkdf2_sha256.hash(otp)
+
+        expiry = datetime.now() + timedelta(hours=4)
+
+        passwordResetEntry = PasswordReset(email=email, one_time_password=hashed_otp, expiry_time=expiry, has_reset=False)
+
+        reset_password_email(otp, user_query[0].email, user_query[0].name)
+
+        session.add(passwordResetEntry)
+        session.flush()
+
+    except Exception as e:
+        print(e)
+        return jsonify({
+            'success': False,
+            'error': 'unknown',
+            'message': str(e),
+        })
+
+    else:
+        session.commit()
+        return jsonify({
+            'success': True,
+            'error': 'none',
+            'message': '',
+        })
+
+
+@authentication.route('/reset-password', methods=["POST"])
+def reset_password():
+    try:
+        request_json = request.get_json()
+
+        email = request_json["email"]
+        new_password = request_json["password"]
+        one_time_password = request_json["otp"]
+         
+        user = session.query(User).filter(User.email == email)[0]
+
+        passwordResetInfo = session.query(PasswordReset).filter(PasswordReset.email == email).order_by(
+            desc(PasswordReset.expiry_time))[0]
+
+        if (passwordResetInfo.expiry_time < datetime.now()):
+            return jsonify({
+                'success': False,
+                'message': 'expired'
+            })
+
+        if not pbkdf2_sha256.verify(one_time_password, passwordResetInfo.one_time_password):
+            return jsonify({
+                'success': False,
+                'message': 'otp'
+            })
+
+        passwordResetInfo.has_reset = True
```

Comment:
I think we need a session.commit() after this

Reply:
Yeah good spot